mlx_graphs/utils/transformations.py

```diff
@@ -155,3 +155,116 @@ def get_src_dst_features(
         )
 
     return src_val, dst_val
+
+
+def get_unique_edge_indices(edge_index_1: mx.array, edge_index_2: mx.array) -> mx.array:
+    """
+    Compute the indices of the edges in edge_index_1 that are NOT present in edge_index_2
+
+    Args:
+        edge_index_1: The first edge index array.
+        edge_index_2: The second edge index array.
+
+    Returns:
+        The indices of the edges in edge_index_1 that are not present in edge_index_2
+
+
+    Example:
+
+    .. code-block:: python
+
+        edge_index_1 = mx.array([
+                                    [0, 1, 1, 2],
+                                    [1, 0, 2, 1],
+                                ])
+        edge_index_2 = mx.array([
+                                    [1, 2, 2],
+                                    [2, 1, 2],
+                                ])
+        x = remove_common_edges(edge_index_1, edge_index_2)
+        # [0, 1]
+    """
+    edge_1_tuples = [tuple(edge.tolist()) for edge in edge_index_1.transpose()]
+    edge_2_tuples = [tuple(edge.tolist()) for edge in edge_index_2.transpose()]
+
+    return mx.array(
+        [i for i, edge in enumerate(edge_1_tuples) if edge not in edge_2_tuples]
```

Comment:
This loop in O(n^2) can be transformed in O(n) with a set:
```python
edge_1_tuples = [tuple(edge.tolist()) for edge in edge_index_1.transpose()]
edge_2_set = set(tuple(edge.tolist()) for edge in edge_index_2.transpose())

return mx.array(
    [i for i, edge in enumerate(edge_1_tuples) if edge not in edge_2_set]
)
```

Reply:
Unfortunately, I'm not aware of an optimal way to vectorize the for loop here even with numpy, but it should be ok even with some millions edges, you can try locally if you want to be sure.

Reply:
Yes, good catch!